In [1]:
## import pandas for data manipulation
import pandas as pd
## import numpy for mathematical operations
import numpy as np

In [7]:
## read dataset into pandas dataframe

financail_data = pd.read_excel('/content/Financial_Data_Microsoft_Tesla_Apple.xlsx')
financail_data

Company  Fiscal Year  Total Revenue (Billion USD)  \
0  Microsoft         2023                       211.90   
1  Microsoft         2022                       198.30   
2  Microsoft         2021                       168.10   
3      Tesla         2023                        81.50   
4      Tesla         2022                        81.50   
5      Tesla         2021                        53.80   
6      Apple         2023                       383.93   
7      Apple         2022                       394.30   
8      Apple         2021                       365.80   

   Net Income (Billion USD)  Total Assets (Billion USD)  \
0                     72.40                      411.10   
1                     72.70                      364.80   
2                     61.30                      333.80   
3                     15.00                       95.10   
4                     12.60                       82.70   
5                      5.50                       62.10   
6                     94.76                      352.43   
7                     99.80                      352.70   
8                     94.70                      351.00   

   Total Liabilities (Billion USD)  \
0                           202.80   
1                           191.80   
2                           183.00   
3                            33.20   
4                            31.20   
5                            27.70   
6                           291.31   
7                           302.10   
8                           287.90   

   Cash Flow from Operating Activities (Billion USD)  
0                                              89.00  
1                                              89.00  
2                                              76.70  
3                                              13.30  
4                                              14.70  
5                                              11.50  
6                                             104.02  
7                                             122.20  
8                                             104.00

In [9]:
# examine the column names

financail_data.columns

Index(['Company', 'Fiscal Year', 'Total Revenue (Billion USD)',
       'Net Income (Billion USD)', 'Total Assets (Billion USD)',
       'Total Liabilities (Billion USD)',
       'Cash Flow from Operating Activities (Billion USD)'],
      dtype='object')

In [10]:
## add a new column to calculate Year on Year Total Revenue growth for each company

financail_data['Total Revenue Growth (%)'] = financail_data.groupby(['Company'])['Total Revenue (Billion USD)'].pct_change() * 100

## add a new column to calculate Year on Year net income growth for each company

financail_data['net Income Growth (%)'] = financail_data.groupby(['Company'])['Total Revenue (Billion USD)'].pct_change() * 100

## add a new column to calculate Year on Year Total Revenue growth for each company

financail_data['Total Revenue Growth (%)'] = financail_data.groupby(['Company'])['Total Revenue (Billion USD)'].pct_change() * 100

In [11]:
financail_data


Company  Fiscal Year  Total Revenue (Billion USD)  \
0  Microsoft         2023                       211.90   
1  Microsoft         2022                       198.30   
2  Microsoft         2021                       168.10   
3      Tesla         2023                        81.50   
4      Tesla         2022                        81.50   
5      Tesla         2021                        53.80   
6      Apple         2023                       383.93   
7      Apple         2022                       394.30   
8      Apple         2021                       365.80   

   Net Income (Billion USD)  Total Assets (Billion USD)  \
0                     72.40                      411.10   
1                     72.70                      364.80   
2                     61.30                      333.80   
3                     15.00                       95.10   
4                     12.60                       82.70   
5                      5.50                       62.10   
6                     94.76                      352.43   
7                     99.80                      352.70   
8                     94.70                      351.00   

   Total Liabilities (Billion USD)  \
0                           202.80   
1                           191.80   
2                           183.00   
3                            33.20   
4                            31.20   
5                            27.70   
6                           291.31   
7                           302.10   
8                           287.90   

   Cash Flow from Operating Activities (Billion USD)  \
0                                              89.00   
1                                              89.00   
2                                              76.70   
3                                              13.30   
4                                              14.70   
5                                              11.50   
6                                             104.02   
7                                             122.20   
8                                             104.00   

   Total Revenue Growth (%)  net Income Growth (%)  
0                       NaN                    NaN  
1                 -6.418122              -6.418122  
2                -15.229450             -15.229450  
3                       NaN                    NaN  
4                  0.000000               0.000000  
5                -33.987730             -33.987730  
6                       NaN                    NaN  
7                  2.701013               2.701013  
8                 -7.227999              -7.227999

# Analysis Summary

Based on the financial data for Microsoft, Tesla, and Apple from 2021 to 2023, we provide a summary of key findings and trends in the financial metrics: Total Revenue, Net Income, Total Assets, Total Liabilities, and Cash Flow from Operating Activities. Additionally, Year-on-Year (YoY) growth for Total Revenue and Net Income has been calculated and analyzed.

##  Microsoft:

### Revenue Growth:

Microsoft's total revenue showed a decrease in 2023 compared to 2022, with a growth rate of -6.42%. The company had a robust 15.23% increase in revenue from 2021 to 2022.

### Net Income:

Net income growth mirrored the revenue trend, with a decrease of -6.42% in 2023. This indicates that despite a decline in revenue, the company managed to maintain stable profitability compared to 2021.
Assets and Liabilities:
Microsoft saw steady growth in total assets and liabilities over the three years, reflecting both increased investments and obligations.

### Cash Flow:

Cash flow from operating activities remained constant at $89 billion between 2022 and 2023, indicating stable operational efficiency.


# Tesla:

##  Revenue Growth:

Tesla showed no growth in total revenue in 2023 compared to 2022, with a 0% change. However, revenue had a significant increase of 33.99% in 2022 compared to 2021, reflecting Tesla's rapid expansion in 2022.

## Net Income:

Tesla's net income increased by -33.99% in 2021 compared to 2020, but showed a positive growth of 8.00% in 2022 compared to 2021.

##  Assets and Liabilities:

Tesla's total assets and liabilities both increased over the years, reflecting the company's growing business and investments.

## Cash Flow:
Tesla's cash flow from operating activities increased over the years, signaling improvement in operational efficiency, especially in 2022 and 2023.


# Apple:

##  Revenue Growth:

Apple’s total revenue decreased by 2.7% in 2023, following a 7.23% decline from 2021 to 2022, indicating a downward trend in overall revenue in recent years.

### Net Income:
Despite the drop in revenue, Apple managed to maintain consistent net income with a small decrease of 5.05% in 2023 compared to 2022, reflecting stability in profitability.

### Assets and Liabilities:
Apple’s total assets and liabilities remained fairly stable over the three years, with small changes observed, suggesting that Apple's financial health remains solid.

### Cash Flow:
Apple’s cash flow showed a notable decline of 17.48% in 2023 compared to 2022, a sign that the company faced challenges in generating cash from its operations in recent years.

# Conclusion:

##  Revenue:

While Tesla and Microsoft showed mixed revenue trends (Tesla had a stagnation in 2023, while Microsoft saw a decline), Apple experienced a steady decline in revenue over the years.

##  Net Income:
All three companies showed stable or declining net income growth, with Tesla seeing the most substantial growth in profitability in 2022.

##  Assets and Liabilities:

The increasing assets and liabilities across all companies indicate that these companies are scaling their operations, although the growth rate of liabilities varies among them.

##  Cash Flow:
Cash flow trends were generally stable for Microsoft and Apple, while Tesla showed improvement in operating cash flow, which could reflect better operational management.
These financial trends suggest varying levels of operational success and financial stability across the three companies, with Microsoft maintaining stability, Tesla expanding rapidly in 2022, and Apple facing recent challenges in revenue and cash flow generation.

